validation/standard/versions.go

```diff
@@ -1,13 +1,137 @@
 package standard
 
 import (
+	"errors"
+	"fmt"
 	"reflect"
+	"strings"
 
 	"github.com/ethereum-optimism/superchain-registry/superchain"
+	"golang.org/x/mod/semver"
 )
 
 type Tag string
 
+type MappedContractProperties[T string | VersionedContract] struct {
+	L1CrossDomainMessenger       T `toml:"l1_cross_domain_messenger,omitempty"`
+	L1ERC721Bridge               T `toml:"l1_erc721_bridge,omitempty"`
+	L1StandardBridge             T `toml:"l1_standard_bridge,omitempty"`
+	L2OutputOracle               T `toml:"l2_output_oracle,omitempty"`
+	OptimismMintableERC20Factory T `toml:"optimism_mintable_erc20_factory,omitempty"`
+	OptimismPortal               T `toml:"optimism_portal,omitempty"`
+	OptimismPortal2              T `toml:"optimism_portal2,omitempty"`
+	SystemConfig                 T `toml:"system_config,omitempty"`
+	// Superchain-wide contracts:
+	ProtocolVersions T `toml:"protocol_versions,omitempty"`
+	SuperchainConfig T `toml:"superchain_config,omitempty"`
+	// Fault Proof contracts:
+	AnchorStateRegistry     T `toml:"anchor_state_registry,omitempty"`
+	DelayedWETH             T `toml:"delayed_weth,omitempty"`
+	DisputeGameFactory      T `toml:"dispute_game_factory,omitempty"`
+	FaultDisputeGame        T `toml:"fault_dispute_game,omitempty"`
+	MIPS                    T `toml:"mips,omitempty"`
+	PermissionedDisputeGame T `toml:"permissioned_dispute_game,omitempty"`
+	PreimageOracle          T `toml:"preimage_oracle,omitempty"`
+	CannonFaultDisputeGame  T `toml:"cannon_fault_dispute_game,omitempty"`
+}
+
+// ContractBytecodeHashes stores a bytecode hash against each contract
+type ContractBytecodeHashes MappedContractProperties[string]
+
+// VersionedContract represents a contract that has a semantic version.
+type VersionedContract struct {
+	Version string `toml:"version"`
+	// If the contract is a superchain singleton, it will have a static address
+	Address *superchain.Address `toml:"implementation_address,omitempty"`
+	// If the contract is proxied, the implementation will have a static address
+	ImplementationAddress *superchain.Address `toml:"address,omitempty"`
```

Comment:
The struct tags don't look quite right, here.

Reply:
You are correct, although this pr did not introduce those tags. Rather they were just moved from their previous file so the mismatch already existed. That has since been fixed in https://github.com/ethereum-optimism/superchain-registry/pull/739